File: client/src/models/user/UserInfoViewModel.ts
import { z } from 'zod';

const UserInfoViewModel = z.object({
  id: z.string(),
  userName: z.string(),
  userImageUrl: z.string().nullable(),
  aboutMe: z.string().nullable(),
  isCritic: z.boolean(),
});

export default UserInfoViewModel;
export type UserInfoViewModelType = z.infer<typeof UserInfoViewModel>;

File: Server/src/Server.Application/ViewModels/UserInfoViewModel.cs
namespace Server.Application.ViewModels;

public sealed class UserInfoViewModel
{
    public Guid Id { get; set; }
    public required string UserName { get; set; }
    public string? UserImageUrl { get; set; }
    public bool IsCritic { get; set; }
    public string? AboutMe { get; set; }
}